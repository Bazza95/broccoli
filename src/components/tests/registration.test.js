import React from 'react';
import Registration from '../registration';
import Text from '../shared/text';
import CircularProgress from '@material-ui/core/CircularProgress';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { shallow } from 'enzyme';
import {mount} from "enzyme/build/index";


describe("Registration", () => {
    let state;
    let props;
    let dialog;
    const registration = () => {
        dialog = mount(
            <Registration
                {...props}
                pending={state.pending}
                validation={state.validation}
                validationError={state.validationError}
            />
        );
        return dialog;
    };

    beforeEach(() => {
        state = {
            pending: undefined,
            validation: undefined,
            validationError: undefined
        };
        props = {
            open: true
        };
        dialog = undefined;
    });

    describe("`Registration` renders", () => {
        it('renders as expected', () => {
            const wrapper = shallow(
                <Registration open={true}/>
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    // When the pending prop is true, CircularProgress is rendered
    describe("when `pending` is true", () => {
        beforeEach(() => {
            state.pending = true;
        });

        it("renders `CircularProgress`", () => {
            expect(registration().find(CircularProgress));
        });
    });

    // When the validation prop is false, the form should render - containing 3 text fields
    describe("when `validation` is false", () => {
        beforeEach(() => {
            state.validation = false;
        });

        it("renders `Text`", () => {
            expect(registration().find(Text).length).toBe(3);
        });
    });

    // When the pending and validation states are undefined, the button text should be "Send"
    describe("when `validation` and 'pending' are undefined", () => {
        beforeEach(() => {
            state.validation = undefined;
            state.pending = undefined;
        });

        it("renders Send", () => {
            expect(registration().find(Button).text()).toEqual("Send");
        });
    });

    // When the validationError state is not an empty string and validation state is false,
    // a Typography component should render
    describe("when `validationError` is not empty string and validation is false", () => {
        beforeEach(() => {
            state.validationError = "Failed";
            state.validation = false;
        });

        it("renders 'Typography'", () => {
            expect(registration().find(Typography).length).toBe(1);
        });
    });

    // When the validation state is true, a Typography component with component 'p' should render
    describe("when `validation` is true", () => {
        beforeEach(() => {
            state.validation = true;
        });

        it("renders 'Typography' with a p component prop", () => {
            const typography = registration().find(Typography);
            expect(typography.props().component).toEqual("p");
        });
    });

    it("renders `DialogActions`", () => {
        expect(registration().find(DialogActions).length).toBe(1);
    });

});