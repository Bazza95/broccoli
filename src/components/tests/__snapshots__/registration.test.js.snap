// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration \`Registration\` renders renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Registration)
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Registration-button-1",
        "progress": "Registration-progress-2",
      },
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Registration-button-1",
          "progress": "Registration-progress-2",
        },
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
